File: lib/models/component_fields/blocky/data/blocks/math/block_delta_time.dart
import 'package:plock_mobile/models/component_fields/blocky/custom_block.dart';
import '../../../toolbox_block.dart';

/// Custom : Object this block.
final blockDeltaTime = ToolboxBlock(type: "delta_time");

/// Custom : Object this block data.
final blockDeltaTimeJson = CustomBlock.fromJson(
    {
      "type": "delta_time",
      "message0": "delta time",
      "inputsInline": true,
      "output": "Number",
      "colour": 230,
      "tooltip": "Return the delta time",
      "helpUrl": ""
    }
).setFunctionLua('''
  var result = 'deltaTime()';
  return [result, lua.Order.NONE];
''').setFunctionDart('''return ['deltaTime()', Blockly.JavaScript.ORDER_NONE];''')
    .setFunctionJs('''
  return ['deltaTime()', Blockly.JavaScript.ORDER_NONE];
''').setFunctionPhp('''
  return ['deltaTime()', Blockly.JavaScript.ORDER_NONE];
''').setFunctionPython('''
  return ['deltaTime()', Blockly.JavaScript.ORDER_NONE];
''');

File: assets/blockly/blocks/block_delta_time.js
"use strict";
const type_dt = 'delta_time';
Blockly.defineBlocksWithJsonArray([
    {
        "type": type_dt,
        "message0": "deltatime %1",
        "args0": [
            {
                "type": "input_dummy",
            }
        ],
        "output": "Number",
        "colour": 230,
        "tooltip": "get delta time",
        "helpUrl": ""
    }
]);
javascript.javascriptGenerator.forBlock[type_dt] = function (block, generator) {
    return [`sendMessage("deltaTime", JSON.stringify([]))`, javascript.Order.ATOMIC];
};
dart.dartGenerator.forBlock[type_dt] = function (block, generator) {
    return [`deltaTime()`, dart.Order.ATOMIC];
};
lua.luaGenerator.forBlock[type_dt] = function (block, generator) {
    return [`deltaTime()`, lua.Order.ATOMIC];
};
php.phpGenerator.forBlock[type_dt] = function (block, generator) {
    return [`deltaTime()`, php.Order.ATOMIC];
};
python.pythonGenerator.forBlock[type_dt] = function (block, generator) {
    return [`deltaTime()`, python.Order.ATOMIC];
};
